authz/sdk_server_interceptors.go

```diff
@@ -140,12 +141,14 @@ func (i *FileWatcherInterceptor) updateInternalInterceptor() error {
 	if bytes.Equal(i.policyContents, policyContents) {
 		return nil
 	}
-	interceptor, err := NewStatic(string(policyContents))
+	policyContentsString := string(policyContents)
+	interceptor, err := NewStatic(policyContentsString)
 	if err != nil {
 		return err
```

Comment:
`i.policyContents = policyContents` or else this will lead to log spam.

Or right after the `if bytes.Equal`, do `defer func() { i.policyContents = policyContents }()` and remove line 150.

Reply:
Oh I see, we want to avoid logging repetitive error messages. Updated code to log only once. Additionally this optimizes code for cases where same invalid policy appears across multiple reload, then we attempt to do the translation to rbac every time. So now i.policyContents will be the contents of the file (valid/invalid). I added an additional variable latestValidPolicyContents, which points to the policy that we use to make authorization decisions. Its currently not really used, since we have created the engines, and we just use that. But I still thot it may be worthwhile to keep it. Let me know if you think we should just remove it.

Reply:
Oh yeah, no reason to `defer`.  😳 

I would remove `latestValidPolicyContents` until it's needed.  There's reason to have unused code and memory.

Reply:
Done.